### DISCUSSION TITLE
Funding of organization

### DISCUSSION THREAD

**Discussion by deajan** [TriliumNext/Notes#27] [General]
So this might come perhaps too quickly, but I think it's important to talk about it.

I really want this project to succeed. I actually "sold" the idea of Trilium to alot of people. I even host a couple of Trilium instances for free for friends / collegues since I'm really convinced of the solution.
I think one of the good ideas would be to fund the organization, so developpers can share a bit of "coffee / beer money", perhaps even one day employ a part time developper.

I am already funding projects on Patreon (bcachefs, sandboxie...), and am willing to contribute to whatever project makes my day.
Since I can't contribute in terms of code, I can at least contribute in terms of "coffee" (caffeine being transformed into SLOCs probably).
Perhaps there are others like me that can't contribute to code, but would like to contribute anyway.

**Comment by meichthys** [comment]
I'm all for this but i also would like to ensure that ALL TriliumNext features remain fully free and open (just like Zadam gifted this to us).

**Comment by deajan** [comment]
I share your opinion, TriliumNext should stay fully free without any enterprise BS ^^
I've myself seen too much nice open source projects shifting into community / enterprise editions, where community edition gets plundered year by year.

We could perhaps come up with some kind of agreement that prevents TriliumNext to ever make non open source decisions.

Still, we should provide a way for the developpers get at least some kind of compensation for their free time, being it by donating coffee money, recurrent donations (that's what I am aiming to do), or even developpers wishlist fullfilment, whatever floats the boats.

**Reply by eliandoran** [reply]
@deajan , yeah, I think it's a good idea to have some kind of funding going on after we become more productive. I think as a first step we could create a Patreon and GitHub Sponsors.

**Reply by meichthys** [reply]
Related issue: #127

**Comment by meichthys** [comment]
Funding could also come in handy if we ever wanted to use some of the 'non-free' ckeditor features. I'm not sure about the pricing and licensing though, and I'd prefer to stick with entirely open source solutions if possible. Nevertheless, some non-free ckeditor plugins that have been discussed in the past are:
- https://github.com/zadam/trilium/issues/3112#issuecomment-1236409801
- https://github.com/zadam/trilium/issues/1841#issuecomment-821962559
- https://github.com/zadam/trilium/issues/3332#issuecomment-1321261955
- https://github.com/zadam/trilium/issues/4073#issuecomment-1661441312

**Reply by deajan** [reply]
That wouldn't be funding, but merely a paid version, since we cannot just "buy an unlimited ckeditor license" to distribute freely as far as my understandings are going.

**Reply by meichthys** [reply]
Ah ok, i was assuming there would be a license that would allow an open source project to use the features.

**Reply by eliandoran** [reply]
@meichthys , I thought about this too. I suppose paid features would not be possible in an open-source project since we would also make the source code of the plugins public. However, that does not mean we cannot try to negotiate with the CKEditor developers, they might be inclined to offer us a solution even for an open-source project.

**Reply by ProfDoof** [reply]
One thing I noted about ckeditor is that it does appear that you can create your own widgets (though I'm not sure how efficient that is) so we could also look at adding our own widgets or adding support for adding widgets to the ckeditor instance through the plugins system.

**Reply by meichthys** [reply]
CKEditor Open Source License reference: https://ckeditor.com/wysiwyg-editor-open-source/

**Reply by zerebos** [reply]
That's an interesting find @meichthys

**Comment by rmkni** [comment]
I suggest establishing a Trilium collective on the hosting platform https://opencollective.com/opensource, which is widely used by numerous open source projects. This platform allows for easy and transparent donations, with a 10% commission fee.

**Reply by deajan** [reply]
10% seems a bit rough... How about https://liberapay.com/ which takes 3-5% depending on the payment system ?
Perhaps even github sponsor that takes 6%. Even Patreon only takes 8%.

After having read a bit about opencollective, I don't find the 10% commission fee. As far as I can read, only the fiscal host pays 15% fees on it's invoices, which should be fairly low. Where did you read about the 10% ?

**Reply by meichthys** [reply]
It's found here. It appears that most FOSS software funded by OpenCollective is funded by the "Open Source Software" collective (which is just one of many 'collectives' on OpenCollective) - very confusing naming conventions there, but here's the page for the OpenSourceSoftware Collective:
https://opencollective.com/opensource

I do like the Liberapay option too, but one-time donations aren't as straightforward there. Last time i checked, you had to setup a recurring donation and then cancel it, or set a limit to prevent the recurrence.

**Comment by rmkni** [comment]
[Fee are 10% of incoming funds, plus payment processor fees. These cover our operating costs and support our initiatives to strengthen the open source ecosystem.](https://www.oscollective.org/#:~:text=Fee%20are%2010%25%20of%20incoming%20funds%2C%20plus%20payment%20processor%20fees.%20These%20cover%20our%20operating%20costs%20and%20support%20our%20initiatives%20to%20strengthen%20the%20open%20source%20ecosystem.)

10% may be higher compared to others.
However, I think we should consider open collective advantages:
- it operates [transparently](https://opencollective.com/opensource#category-BUDGET) and the trilium community could operates transparently too
- the platform could bring visibility to the project and bring new users, contributors and donators
- we can ask for help if we need guidance for projects, funding, growing etc.

**Comment by Alumniminium** [comment]
How about we do issue bounties? People can put up money for issues and the dev(s) that close it get it?

**Reply by rmkni** [reply]
I like the idea but how could we organize this?

**Reply by Alumniminium** [reply]
https://gitpay.me something like that?

**Reply by en3r0** [reply]
I am a big fan of that approach. Helps work get done and paid for.

**Reply by rmkni** [reply]
I had trouble registering, I don't receive the verification mail, there are 10 issues with bounties open since more than a year, it seems not active

**Reply by rmkni** [reply]
With open collective, it is possible to create "projects" and have people donate for these projects specifically, could be a solution to have the money prioritize the work maybe?

**Comment by CobriMediaJulien** [comment]
Why not funding the project with easy trilium hosting for lazy people like me? I think this would be fair. Right now i host it on my own server but i would love to spend some money for a "managed solution". The normal user could use trilium for some money per month or year. If u host it yourself it´s completely free. Would make trilium available for less technical expierenced users too.

**Reply by meichthys** [reply]
This is a good option, but it can be a slippery slope too. I've seen far too many projects do this and then start offering other "services" on their hosted instances for a small fee, and before long there is no longer feature parity between the hosted and self hosted options. 

I'm all for a hosted option as long as we can ensure the self hosted option can maintain feature parity (without significant effort).

@CobriMediaJulien what do you think would be a reasonable monthly fee for a hosted instance for say up to 5gb storage?

Also, @sigaloid how do you handle scripting on your instances? It send to me that scripting could be abused (ie. Crypto mining or other more malicious scripts)

**Reply by deajan** [reply]
@meichthys 
I'm currently hosting multiple instances of Trilium for my own organization and some others.
At my current pricing, a Trilium instance with 5GB of storage, including GFS backups (30 days, 4 weeks, 12 months, 2 years), geo-replicated backups and security layers would cost roughly 1,7€ / month (buying price).
The only thing that has yet to be determined is the upgrade policy, which could change the pricing a bit, because of laber frequency (including rollbacks).

As per security, my datacenter offerings are ANSSI-BP-028 security enhanced. All instances are isolated of course, and can be individually updated on request, or left with a specific version unless a security issue raises.
For your crypto mining question... AFAIK the server only hosts, but doesn't execute scripts, which are all executed by the client. Is there anything I didn't understand in Trilium's arch ?

I am willing to make this offer to the Trilium / TriliumNext community, eg hosted Trilium at buyer's price. Would be kind of my contrib to the organization.

There would be only two limits for me:
- Limited to EU market, because of regulations (our hosting is RGPD compliant, and I guess there are enough international hosting players for the world :)
- Limited to one offer per physical / moral persona

What do you think ?

**Reply by meichthys** [reply]
> AFAIK the server only hosts, but doesn't execute scripts, which are all executed by the client. Is there anything I didn't understand in Trilium's arch ?

The api supports a "runOnBackend" method, but I'm not familiar with how it works. I do know some existing plugins use it.

**Reply by deajan** [reply]
Well then. There's no "good way" to limit bad scripts IMO, except the basics:
- Not allowing outgoing connections from the backend (limits sending data to third party)
- Filtering on the reverse proxy (bad reputation IP lists, WAF)
- Security limits on instance

It's almost the same as any hosting service, customers can exec bad scripts. This case is still easier since Trilium isn't supposed to initiate connections outside, except for update queries.

**Reply by meichthys** [reply]
I'm not too sure how important the run-on-backend API method is. It may be possible for us just to drop that method and rely on client-side scripts for hosted instances. However, that would require more investigation.

**Reply by CobriMediaJulien** [reply]
@meichthys @deajan 
To be honest, this was just an idea so people "donate" more, which could be invested in the development of the app. For a normal user, it's not usual to donate to a open source development via a donation because he doesn't know the skills and time the team spents in development. If u give them some kind of small service back, even if it´s just some small stuff, then more people are willing to support you. Imho just be transparent and there won't be trouble. And let the community participate and vote. You can still decide, but participation and transparency are key.

And by the way, there are allready option that charge for hosting. I just found this: https://trilium.cc

Look, if u give people a small service they are more likely and willing to support you with their money. This would enable the organisation to develop trilium further e.g. with bug fixing competitions and so on. You could show a transparent report every 4 month about what you earned and how you spend it on triliumnext.

It´s just human behavior, and u feel better if u get sth. back for your money. Of course trilium is for free but just donations is too much of a hurdle for somebody who doesn´t know how much work the app is.

I don´t think u risk feature parity if you do it correctly. One good example is nextcloud. I spent so much time that my nextcloud is finally working in my local network. I hated it. Wouldn´t mind after this journey saving some time and spending some money in nextcloud hosting and through that, supporting directly the development of the application. Everything around 5 Euro is fine for me, you can also ask the community in some month and let them decide. Just give the tech guys the option to self-host and give the non-tech guys like me who want to save some time the option to support and use this awesome app hustle free with all features like synching several devices.

Another idea would be to croudfound every bigger featurerequest. People vote with their "bids" and everyone can give what he thinks the feature is worth for him. Then you know the most expected features, give the community to support without spending a lot of time or having certain skills and last but not least influence Trilium in the way they like. Every feature development will be reported back to the community transparently. I could help with reports and organisation of this stuff.

Features that aren´t implemented just receive a refund. I´m still brainstorming how i can support trilium without having years of coding expierence.

Regarding your cyber sec issue:
In case you need a pen tester, maybe i can ask some guys for a favor.

**Reply by sigaloid** [reply]
@meichthys I have monitoring on resource usage which has helped me identify bugs in the past w.r.t. upgrades failing and churning CPU forever. But I've never had any serious problems with this, as everything is kept isolated in containers

**Reply by sigaloid** [reply]
Just had this thought - I would recommend to contact PikaPods to allow them to host TriliumNext instances - they can automatically contribute a percentage back to the OSS project, and they've utilized scaling many instances to offer extremely cheap instances.

**Comment by en3r0** [comment]
I would like to switch my sponsorship over to this github organization whenever someone sets it up.

**Reply by eliandoran** [reply]
@en3r0 , appreciated. However let's keep this for when we will provide value, or more specifically after we deliver our first beta version.

**Reply by en3r0** [reply]
Sounds good, I will be on the lookout for it!

**Reply by eliandoran** [reply]
@en3r0 , if you're still using TriliumNext, it's now available: https://github.com/sponsors/eliandoran

**Comment by St33f3n** [comment]
Thanks so much for the recent updates to TriliumNext! I'm really enjoying some features of the updates. They're making a difference. I'd love to support the continued development of this fantastic app and its commitment to remaining "libre" software. Do you have any updates regarding donation?

**Reply by CobriMediaJulien** [reply]
I´m not aware. @eliandoran has to decide this in my eyes. I would like to support too and maybe now is a good time after the community and elian is so active.

**Reply by eliandoran** [reply]
@St33f3n , @CobriMediaJulien , thank you for your interest.

I've set up a personal GitHub sponsors for now, if you would like to sponsor me directly: https://github.com/sponsors/eliandoran

I'll also leave a more detailed reply below.

**Comment by PawsFunctions** [comment]
I am not affiliated to futo, They gave some funds to [Immich](https://github.com/immich-app/immich). TriliumNext is good project, they may provide some grant or help in future development. They want/love AGPL license which TriliumNext uses so it should be easy for them to support.
 
https://futo.org/about/what-is-futo/

https://www.youtube.com/watch?v=uyTPqxgqgjU

**Reply by eliandoran** [reply]
@PawsFunctions , if I remember correctly, the story of Immich was that it was vouched for by Louis Rossmann (and the link you have provided is also indicative of it) and we're nowhere near as popular and we are only starting out.

Might be interesting to consider them for the future.

**Comment by eliandoran** [comment]
Donations are organization are a little bit more difficult to plan, so in the meantime I've decided (but also asking a few people around) to set up a GitHub Sponsors on my personal account.

If you would like to sponsor me, I would appreciate it and it would help keep me motivated to do work on the project: https://github.com/sponsors/eliandoran

**Reply by meichthys** [reply]
It is with very great pleasure that I have become Elian's first sponsor!
![Screenshot_20241217-181255](https://github.com/user-attachments/assets/0d6516e7-b33d-4ea9-b2bd-8bf604400a8c)

**Reply by maphew** [reply]
seconded! 
And shouted out too: https://vmst.io/@maphew/113945753247169575

**Reply by meichthys** [reply]
@MarcS1975, No pressure or anything, just wanted to let you know (Since i saw [your post over on Trilium](https://github.com/zadam/trilium/issues/4620#issuecomment-2642832898)) that - if you wanted to sponsor TriliumNext, [Elian's link above](https://github.com/sponsors/eliandoran) is how you could do it :)

**Comment by Disentimes** [comment]
> So this might come perhaps too quickly, but I think it's important to talk about it.
> 
> I really want this project to succeed. I actually "sold" the idea of Trilium to alot of people. I even host a couple of Trilium instances for free for friends / collegues since I'm really convinced of the solution. I think one of the good ideas would be to fund the organization, so developpers can share a bit of "coffee / beer money", perhaps even one day employ a part time developper.
> 
> I am already funding projects on Patreon (bcachefs, sandboxie...), and am willing to contribute to whatever project makes my day. Since I can't contribute in terms of code, I can at least contribute in terms of "coffee" (caffeine being transformed into SLOCs probably). Perhaps there are others like me that can't contribute to code, but would like to contribute anyway.
Hello,

I can't code to save my life and financially indisposed but I still want to help as a way to give back to this great community and amazing project.

I would propose another way to help funding the project: Merchandise (aka Merch for people born after 2005). Many people give way more when they get some trinket in return.

- T-shirt , cups, woodies  with the Trilium-Next logo 
- Physical notebook with the Trilium-Next logo
- limited edition Trilium-Next merch with the different versions , right now it would be 0.91.6 , but I can see people collecting those.

You can find countless merchandise printing websites on the net, but try to find one trustworthy which give good value and also send the merchandise outside of North-America for us outside of the American Dream.

**Comment by MarcS1975** [comment]
> I really want this project to succeed. I actually "sold" the idea of Trilium to alot of people. I even host a couple of Trilium instances for free for friends / collegues since I'm really convinced of the solution.

I am very grateful for the team that is picking this amazing poject up and of course to the original initiator. I love this app and the fact that it can be entirely self-hosted makes this a very worthwhile project. Unfortunately I cannot code myself but I would support any ideas you come up with to help your work. I want to support maintaining and enhancing Trilliumnext, be it by providing some financial support (if that helps at this stae) or by providing time spent on writing bug reports and input on features. 

I consider this already a very professionally put together application so every additional effort is worth supporting. We really have something amazing at our hands here, and if we can find the resources to make it better, it will be amazing. I saw the link above for sponsorship.

**Comment by eliandoran** [comment]
Closing for now as we have already introduced a way to donate.
